scripts/!mods_preload/mod_reforged.nut

```diff
@@ -44,21 +44,18 @@ foreach (requirement in requiredMods)
 	::Reforged.HooksMod.hookTree("scripts/entity/tactical/actor", function(q) {
 		q.onDeath = @(__original) function( _killer, _skill, _tile, _fatalityType )
 		{
-			if (_killer != null && _killer.getFaction() != ::Const.Faction.Player && _killer.getFaction() != ::Const.Faction.PlayerAnimals)
-			{
-				local playerRelevantDamage = 0.0;
-				if (::Const.Faction.Player in this.m.RF_DamageReceived)
-					playerRelevantDamage += this.m.RF_DamageReceived[::Const.Faction.Player].Total;
-				if (::Const.Faction.PlayerAnimals in this.m.RF_DamageReceived)
-					playerRelevantDamage +=  this.m.RF_DamageReceived[::Const.Faction.PlayerAnimals].Total;
-
-				// If _killer isn't already a player character or player animal AND the player + player animals did at least 50%
-				// of total damage to this actor, we set the _killer to null to ensure that the loot properly drops from this actor.
-				// This is because vanilla drops loot if _killer is null or belongs to Player or PlayerAnimals faction
-				// Warning: This will break any mod that hooks the original onDeath and expects _killer to represent the actual killer
-				if (playerRelevantDamage / this.m.RF_DamageReceived.Total >= 0.5)
-					_killer = null;
-			}
+			local playerRelevantDamage = 0.0;
+			if (::Const.Faction.Player in this.m.RF_DamageReceived)
+				playerRelevantDamage += this.m.RF_DamageReceived[::Const.Faction.Player].Total;
+			if (::Const.Faction.PlayerAnimals in this.m.RF_DamageReceived)
+				playerRelevantDamage +=  this.m.RF_DamageReceived[::Const.Faction.PlayerAnimals].Total;
+
+			// If _killer isn't already a player character or player animal AND the player + player animals did at least 50%
+			// of total damage to this actor, we set the _killer to null to ensure that the loot properly drops from this actor.
+			// This is because vanilla drops loot if _killer is null or belongs to Player or PlayerAnimals faction
+			// Warning: This will break any mod that hooks the original onDeath and expects _killer to represent the actual killer
+			if (playerRelevantDamage / this.m.RF_DamageReceived.Total >= 0.5)
+				_killer = null;
```

Comment:
This revert still doesn't fix the issue where the player can kill-steal and cheese early items.

For that to fix you need to change the _killer from being a player or `null` to some dummy entity from some dummy faction, when the player did **less** than 50% damage.

Reply:
I'd say if the player didn't do 50% damage we can set the `_killer` to the dying entity himself. Would that be ok?

Reply:
Sounds good if that actually works.

Iirc in vanilla the loot is assigned to the player if an enemy dies to damage over time effects.
So in Reforged we need to intervene in this exact scenario so that the items are awared to the player **only** if he did 50%+ damage. And this part I'm missing atm

Reply:
That part will be handled if we set every dying entity's killer to the entity himself if the player+player animals did less than 50% damage. So I will make this change.